modules/core/src/main/java/org/apache/ignite/IgniteCache.java

```diff
@@ -136,6 +136,28 @@
      */
     public IgniteCache<K, V> withPartitionRecover();
 
+    /**
+     * Gets an instance of {@code IgniteCache} that will perform backup nodes check on each get attempt.
+     *
+     * Read Repair means that each backup node will be checked to have the same entry as primary node has,
+     * and in case consistency violation found:
+     * for transactional caches:
+     * - values across the topology will be replaced by latest versioned value:
+     * -- automaticaly for OPTIMISTIC || READ_COMMITTED transactions
+     * -- at commit() for PESSIMISTIC && !READ_COMMITTED transactions
+     * - consistency violation event will be recorded in case it's configured as recordable
+     * for atomic caches: consistency violation exception will be thrown.
+     *
+     * One more important thing is that this proxy usage does not guarantee "all copies check" in case value
```

Comment:
> this proxy usage does not guarantee "all copies check" in case value already cached inside the transaction.

Can't we invalidate cache for this case?

Reply:
tx visibility can't be hacked.